dns/examples/mdns.rs

```diff
@@ -0,0 +1,65 @@
+//! This is a simulation of the W5500 registers that runs on your local machine.
+//!
+//! This does not require any embedded hardware to use.
+//!
+//! **Note:** This will communicate external network services.
+
+use std::time::{Duration, Instant};
+use w5500_dns::{
+    hl::{net::Eui48Addr, Error},
+    ll::{Registers, Sn, VERSION},
+    mdns::Client as MdnsClient,
+    Hostname,
+};
+use w5500_regsim::W5500;
+
+// DNS socket to use, this could be any of them
+const DNS_SOCKET: Sn = Sn::Sn3;
+const DNS_SRC_PORT: u16 = 45917;
+
+const DEFAULT_MAC: Eui48Addr = Eui48Addr::new(0xDE, 0xAD, 0xBE, 0xEF, 0xFE, 0xED);
+
+fn main() {
+    // this enables the logging built into the register simulator
+    stderrlog::new()
+        .verbosity(3)
+        .timestamp(stderrlog::Timestamp::Nanosecond)
+        .init()
+        .unwrap();
+
+    let mut w5500: W5500 = W5500::default();
+    assert_eq!(w5500.version().unwrap(), VERSION); // sanity check
+
+    w5500.set_shar(&DEFAULT_MAC).unwrap();
+    debug_assert_eq!(w5500.shar().unwrap(), DEFAULT_MAC);
+    log::info!("DEFAULT_MAC = {:x?}", DEFAULT_MAC);
```

Comment:
```suggestion
    log::info!("DEFAULT_MAC = {DEFAULT_MAC}");
```

`Display` should print it in form of `xx:xx:xx:xx:xx:xx`